### PR TITLE
Scheduler: Refactor - try_schedule_transaction

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- Current individual transaction scheduling decision logic is not isolated into a function
- Difficult to make changes to `TransactionStateContainer` to remove fns that **return** mutable access
- Have been trying to make a change which would remove `get_mut_transaction_state` in favor of a function which takes a closure and gives (temporary) mutable access

#### Summary of Changes
- Refactor transaction scheduling into a separate function

Fixes #

### CODE REVIEW

File: core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs [hunk 1]
```diff
@@ -154,62 +156,54 @@ impl PrioGraphScheduler {
...
-                saturating_add_assign!(batches.total_cus[thread_id], cost);
-
-                // If target batch size is reached, send only this batch.
-                if batches.ids[thread_id].len() >= TARGET_NUM_TRANSACTIONS_PER_BATCH {
-                    saturating_add_assign!(num_sent, self.send_batch(&mut batches, thread_id)?);
-                }
+                match maybe_schedule_info {
+                    Err(TransactionSchedulingError::Filtered) => {
+                        container.remove_by_id(&id.id);
+                        continue;
```
Comment (ryoqun) [inline core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs]: i think this `continue` is needless like the `Err(TransactionSchedulingError::UnschedulableConflicts)` arm?

Comment (apfitzge) [reply core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs]: e644ad978d

**Review by ryoqun — APPROVED** [review]
lgtm; nice clean up!

File: core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs [hunk 2]
```diff
@@ -172,26 +174,12 @@ impl PrioGraphScheduler {
...
-                    saturating_add_assign!(num_unschedulable, 1);
-                    continue;
-                }
-
-                // Schedule the transaction if it can be.
-                let transaction_locks = transaction.get_account_locks_unchecked();
-                let Some(thread_id) = self.account_locks.try_lock_accounts(
-                    transaction_locks.writable.into_iter(),
-                    transaction_locks.readonly.into_iter(),
-                    schedulable_threads,
```
Comment (apfitzge) [inline core/src/banking_stage/transaction_scheduler/prio_graph_scheduler.rs:194]: @alessandrod the guy who wrote this is a fucking moron

**Pull request merged**